rye-devtools/src/rye_devtools/find_downloads.py

```diff
@@ -14,388 +13,552 @@
-]
-SPECIAL_TRIPLES = {
-    "macos": "x86_64-apple-darwin",
-    "linux64": "x86_64-unknown-linux-gnu",
-    "windows-amd64": "x86_64-pc-windows-msvc",
-    "windows-x86-shared-pgo": "i686-pc-windows-msvc-shared-pgo",
-    "windows-amd64-shared-pgo": "x86_64-pc-windows-msvc-shared-pgo",
-    "windows-x86": "i686-pc-windows-msvc",
-    "linux64-musl": "x86_64-unknown-linux-musl",
-}
-
-# matches these: https://doc.rust-lang.org/std/env/consts/constant.ARCH.html
-ARCH_MAPPING = {
-    "x86_64": "x86_64",
-    "x86": "x86",
-    "i686": "x86",
-    "aarch64": "aarch64",
-}
-
-# matches these: https://doc.rust-lang.org/std/env/consts/constant.OS.html
-PLATFORM_MAPPING = {
-    "darwin": "macos",
-    "windows": "windows",
-    "linux": "linux",
-}
-
-ENV_MAPPING = {
-    "gnu": "gnu",
-    # We must ignore musl for now
-    # "musl": "musl",
-}
-
-
-@dataclass(frozen=True)
-class PlatformTriple:
+
+class PlatformTriple(NamedTuple):
     arch: str
     platform: str
-    environment: Optional[str]
-    flavor: str
+    environment: str | None
+    flavor: str | None
+
+
+class PythonVersion(NamedTuple):
+    major: int
+    minor: int
+    patch: int
 
     @classmethod
-    def from_str(cls, triple: str) -> Optional[Self]:
-        """Parse a triple into a PlatformTriple object."""
+    def from_str(cls, version: str) -> Self:
+        return cls(*map(int, version.split(".", 3)))
 
-        # The parsing functions are all very similar and we could abstract them into a single function
-        # but I think it's clearer to keep them separate.
-        def match_flavor(triple):
-            for flavor in FLAVOR_PREFERENCES + HIDDEN_FLAVORS:
-                if flavor in triple:
-                    return flavor
-            return ""
+    def __neg__(self) -> Self:
+        return PythonVersion(-self.major, -self.minor, -self.patch)
 
-        def match_mapping(pieces: list[str], mapping: dict[str, str]):
-            for i in reversed(range(0, len(pieces))):
-                if pieces[i] in mapping:
-                    return mapping[pieces[i]], pieces[:i]
-            return None, pieces
 
-        # We split by '-' and match back to front to extract the flavor, env, platform and archk
-        arch, platform, env, flavor = None, None, None, None
+@total_ordering
+class PythonImplementation(StrEnum):
+    # Keep element in the same order as the final download.inc
+    PYPY = "pypy"
+    CPYTHON = "cpython"
 
-        # Map, old, special triplets to proper triples for parsing, or
-        # return the triple if it's not a special one
-        triple = SPECIAL_TRIPLES.get(triple, triple)
-        pieces = triple.split("-")
-        flavor = match_flavor(triple)
-        env, pieces = match_mapping(pieces, ENV_MAPPING)
-        platform, pieces = match_mapping(pieces, PLATFORM_MAPPING)
-        arch, pieces = match_mapping(pieces, ARCH_MAPPING)
+    def __lt__(self, other: Self) -> bool:
+        return self._sort_order_ < other._sort_order_
```

Comment:
I can't find anything about using `_sort_order_` and looking at the source of enum, this seem sto be an internal member. It would be more robust to use the documented guarnatees instead, e,g `str(self) < str(other)`